### ISSUE TITLE
Markdown **emphasis** at the beginning of the line is not rendered.

### ISSUE THREAD

**Issue by dhebbeker** [doxygen/doxygen#8615]
### Describe the bug

Markdown emphasis (using `**word**`) at the beginning of the new line does not render correctly.

For example this code:

```cpp
/*! \file 

not bold

**should be bold**

not bold

this **should be bold**

not bold
*/
```

### Expected behavior
I would expect the following output:

> not bold
> 
> **should be bold**
> 
> not bold
> 
> this **should be bold**
> 
> not bold

Instead the output is:

> not bold
> 
> should be bold**
> 
> not bold
> 
> this **should be bold**
> 
> not bold

### Screenshots
![grafik](https://user-images.githubusercontent.com/1594340/123086082-4adeee80-d423-11eb-80e6-19459eefaa6e.png)

### To Reproduce

Use the code above with default configuration (empty Doxyfile). This is also provided here: [doxy-markdown-emph.zip](https://github.com/doxygen/doxygen/files/6701274/doxy-markdown-emph.zip)

### Version

 - doxygen 1.9.1 (ef9b20ac7f8a8621fcfc299f8bd0b80422390f4b)
 - Windows 10 64 bit

### Additional context

May be related to #7786.

According to the [manual emphasis should be parsed](https://www.doxygen.nl/manual/markdown.html#mddox_emph_spans) if

> it is preceded by a [...] newline

**Comment by albert-github** [comment]
This is not nice, quick and dirty workaround:
```
&zwnj;**should be bold**
```


(note: problem is probably located in the C scanner, scanner.l, as when I use a markdown file it is rendered correctly and ans in version 1.8.3, very old, it is OK)

**Comment by albert-github** [comment]
The mentioned quote:
> it is preceded by a space, newline, or one the following characters <{([,:;

is valid for markdown, but here we have to do with C / Cpp type of code and one of the properties of the C / Cpp comments is that it can have leading asterisks (0 or more) at the beginning of the line that are ignored by doxygen.
In the scanner (`scanner.l`)  this is handled by means by the rule:
```
<DocBlock>^{B}*"*"+/[^/]
```
it could be argued:
- when on previous / next lines there are no leading asterisks these 2 asterisks should be seen as "bold" indicator, but nowhere it is required that the number of asterisks is consistent and also what to do with the, unlikely block:
  ```
  /**
    ** bold item ** text ... <br>
    ** bold item ** text ... <br>
    ** bold item ** text ... <br>
  */
 ```
- when on the same line there is also another set of `**` then the initial `**` should be seen as bold indicators as well, but a construct like:
  ```
  ** text ** bold **
  ```
  would give problems (this is now handled as intended in the snippet). Also the situation that a bold part is spanned over multiple lines would still not be detected.

I'm not sure that there is really a solution for this problem.

**Comment by dhebbeker** [comment]
Maybe a solution could be to document this limitation in the manual, [where the other limitations of emphasis are already described](https://www.doxygen.nl/manual/markdown.html#mddox_emph_spans). As an alternative for that case setting emphasis using `__word__` can be offered, as this works OK.

**Comment by albert-github** [comment]
I've just pushed a proposed, documentation, patch, pull request #8619

**Comment by albert-github** [comment]
Documentation update has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
